modules/iot-core/README.md

```diff
@@ -0,0 +1,112 @@
+# Google Cloud IoT Core Module
+
+This module setup Cloud IoT Core Registry, register IoT Devices and configure Pub/Sub topics required in Cloud IoT Core.
+
+Requires enabling the following APIs:
+* pubsub.googleapis.com
+* cloudiot.googleapis.com
+
+## Simple Example
+
+Basic example showing how to create an IoT Platform (IoT Core), connected to a set of given Pub-Sub topics and provision IoT devices.
+
+Before executing, device certificates shall be created, for example using:
+
+```
+openssl req -x509 -newkey rsa:2048 -keyout rsa_private.pem -nodes -out rsa_cert.pem -subj "/CN=unused"
+```
+
+And then provision public certificate path in the devices yaml file following the convention device_id: device_cert
+
+
+```hcl
+module "iot-platform" {
+  source     = "./iot-core"
+  project_id = "my_project_id"
+  region = "europe-west1"
+  telemetry_pub_sub_topic_id = "telemetry_topic_id"
+  status_pub_sub_topic_id = "status_topic_id"
+  devices_yaml_file = "devices.yaml"
+}
+# tftest:modules=1:resources=2
+
+```
+
+Now, we can test sending telemetry messages from devices to our IoT Platform, for example using the MQTT demo client at https://github.com/googleapis/nodejs-iot/tree/main/samples/mqtt_example
+
+## Example with specific PubSub topics for custom MQTT topics
+
+If you need to match specific MQTT topics (eg, /temperature) into specific PubSub topics, you can use extra_telemetry_pub_sub_topic_ids for that, as in the following example:
+
+```hcl
+module "iot-platform" {
+  source     = "./iot-core"
+  project_id = "my_project_id"
+  region = "europe-west1"
+  telemetry_pub_sub_topic_id = "telemetry_topic_id"
+  status_pub_sub_topic_id = "statu_topic_id"
+  extra_telemetry_pub_sub_topic_ids = [{
+      "mqtt_topic" = "humidity"
+      "pub_sub_topic" =  "hum_topic_id"
+  },
+  {
+      "mqtt_topic" = "temperature"
+      "pub_sub_topic" =  "temp_topic_id"
+  }]
+  devices_yaml_file = "devices.yaml"
+}
+# tftest:modules=1:resources=2
+
+```
+
+## Example integrated with Data Foundation Platform
+In this example, we will show how to extend **[Data Foundations Platform](../../data-solutions/data-platform-foundations/)** including IoT Platform as a new source of data. 
```

Comment:
In this example, we will show how to extend the **[Data Foundations Platform](../../data-solutions/data-platform-foundations/)** to include IoT Platform as a new source of data.